conformance/lang/expressions/unary-expression/unary-minus-expression.balt

```diff
@@ -0,0 +1,1081 @@
+    int:Signed16? d = -32768;
+
+    io:println(-a); // @output 1
+    io:println(-b); // @output 2
+    io:println((-c).toBalString()); // @output ()
+    io:println(-d); // @output 32768
+}
+
+Test-Case: output
+Description: Test unary negative expression for positive optional int:Signed32, no overflow scenarios.
+Labels: DecimalNumber, int:Signed32, nil-literal, numeric-nil-lifting, optional-type, unary-minus, value:toBalString
+
+function init() {
+    int:Signed32? a = 1;
+    int:Signed32? b = 2;
+    int:Signed32? c = ();
+    int:Signed32? d = 2147483647;
+
+    io:println(-a); // @output -1
+    io:println(-b); // @output -2
+    io:println((-c).toBalString()); // @output ()
+    io:println(-d); // @output -2147483647
+}
+
+Test-Case: output
+Description: Test unary negative expression for negative optional int:Signed32, no overflow scenarios.
+Labels: DecimalNumber, int:Signed32, nil-literal, numeric-nil-lifting, optional-type, unary-minus, value:toBalString
+
+function init() {
+    int:Signed32? a = -1;
+    int:Signed32? b = -2;
+    int:Signed32? c = ();
+    int:Signed32? d = -2147483648;
+
+    io:println(-a); // @output 1
+    io:println(-b); // @output 2
+    io:println((-c).toBalString()); // @output ()
+    io:println(-d); // @output 2147483648
+}
+
+Test-Case: output
+Description: Test unary negative expression for positive optional user defined sub types of int,
+              no overflow scenarios.
+Fail-Issue: ballerina-platform/ballerina-lang#32928
+Labels: DecimalNumber, module-type-defn, numeric-nil-lifting, optional-type, singleton-type, union-type, unary-minus, value:toBalString
+
+type Ints -2|-1|0|1|2;
+
+function init() {
+    Ints? a = 1;
+    Ints? b = ();
+    Ints? c = 0;
+
+    io:println(-a); // @output -1
+    io:println((-b).toBalString()); // @output ()
+    io:println(-c); // @output 0
+}
+
+Test-Case: output
+Description: Test unary negative expression for optional user defined sub types of integer built in sub types,
+              no overflow scenarios.
+Fail-Issue: ballerina-platform/ballerina-lang#32928
+Labels: DecimalNumber, module-type-defn, unary-minus, union-type, int:Signed8, int:Unsigned32, numeric-nil-lifting, optional-type
+
+type Ints int:Signed8|int:Unsigned32;
+
+function init() {
+    Ints? a = 12;
+    int:Signed8|int:Unsigned32? b = 12;
+
+    io:println(-a); // @output -12
+    io:println(-b); // @output -12
+}
+
+Test-Case: output
+Description: Test unary negative expression for positive floats, no overflow scenarios.
+Labels: float, DecimalNumber, unary-minus
+
+function init() {
+    float a = 1.0;
+    float b = 101.5;
+    float c = 0;
+    float d = 922439289.0;
+    io:println(-a); // @output -1.0
+    io:println(-b); // @output -101.5
+    io:println(-c); // @output -0.0
+    io:println(-d); // @output -9.22439289E8
+}
+
+Test-Case: output
+Description: Test unary negative expression for negative floats, no overflow scenarios.
+Labels: float, DecimalNumber, unary-minus
```

Comment:
```suggestion
Labels: float, DecimalFloatingPointNumber, unary-minus
```
We need to fix this wherever applicable in both float and decimal tests, it seems. Check in unary plus tests as well.